istanbul/client.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 AMIS Technologies
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package istanbul
+
+import (
+	"context"
+	"math/big"
+	"sort"
+	"strings"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	ethClient "github.com/getamis/eth-client/client"
+)
+
+type client struct {
+	ethClient.Client
+	c *rpc.Client
+}
+
+func Dial(rawurl string) (Client, error) {
+	rc, err := rpc.Dial(rawurl)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &client{
+		Client: ethClient.NewClient(rc),
+		c:      rc,
+	}
+
+	return c, nil
+}
+
+func (c *client) ProposeValidator(ctx context.Context, address common.Address, auth bool) error {
+	var r []byte
+	err := c.c.CallContext(ctx, &r, "istanbul_propose", address, auth)
```

Comment:
we probably should rename it, c.c looks odd

Reply:
i want change to `c.rpc.CallContext(...)` .  WDYT?

Reply:
sounds good to me, @alanchchen / @markya0616 / @tailingchen  ?

Reply:
c.rpc looks good

Reply:
Both are fine with me :)